frame/clearing-house/src/tests/close_market.rs

```diff
@@ -0,0 +1,206 @@
+use crate::{
+	mock::{
+		accounts::ALICE,
+		runtime::{MarketId, Origin, Runtime, TestPallet, Vamm as VammPallet},
+	},
+	tests::{
+		as_balance, get_market, run_to_time, with_market_context, with_trading_context, ExtBuilder,
+		MarketConfig,
+	},
+	Direction::Long,
+	Error,
+};
+use composable_traits::time::DurationSeconds;
+use frame_support::{assert_noop, assert_ok, error::BadOrigin};
+use proptest::prelude::*;
+
+// ----------------------------------------------------------------------------------------------------
+//                                             Unit Tests
+// ----------------------------------------------------------------------------------------------------
+
+#[test]
+fn only_root_can_close_market() {
+	with_market_context(Default::default(), Default::default(), |market_id| {
+		assert_noop!(TestPallet::close_market(Origin::signed(ALICE), market_id, 1), BadOrigin);
+
+		assert_ok!(TestPallet::close_market(Origin::root(), market_id, 1));
+	})
+}
+
+// ----------------------------------------------------------------------------------------------------
+//                                             Prop Compose
+// ----------------------------------------------------------------------------------------------------
+
+const MAX_DURATION_SECONDS: DurationSeconds = 1_000_000_000;
+
+prop_compose! {
+	fn valid_close_times()(close_market in 2..MAX_DURATION_SECONDS)(
+		close_market in Just(close_market),
+		close_position in 1..close_market,
+	) -> (DurationSeconds, DurationSeconds) {
+		(close_position, close_market)
+	}
+}
+
+prop_compose! {
+	fn invalid_close_times()(close_market in 1..(MAX_DURATION_SECONDS / 2))(
+		close_market in Just(close_market),
+		close_position in (close_market + 1)..=MAX_DURATION_SECONDS,
+	) -> (DurationSeconds, DurationSeconds) {
+		(close_position, close_market)
+	}
+}
+
+prop_compose! {
+	fn invalid_open_close_times()(close_market in 10..=MAX_DURATION_SECONDS)(
+		close_market in Just(close_market),
+		open_position in 1..close_market,
+	) -> (DurationSeconds, DurationSeconds) {
+		(open_position, close_market)
+	}
+}
+
+prop_compose! {
+	fn time_a_before_time_b()(time_b in 1..=MAX_DURATION_SECONDS)(time_a in 0..time_b) -> (DurationSeconds, DurationSeconds) {
+		(time_a, time_a)
```

Comment:
Shouldn't it be `(time_a, time_b)` ?

Reply:
something like:

```rust
prop_compose! {
	fn time_a_before_time_b()(time_b in 1..=MAX_DURATION_SECONDS)(time_a in 0..time_b, time_b in Just(time_b)) -> (DurationSeconds, DurationSeconds) {
		(time_a, time_b)
	}
}
```

Reply:
Ohh, what an oversight, thanks!

Reply:
:wink: 
As the checks performed don't make a difference between equal or less, the tests didn't catch it.
Everything looks okay though, just a small (but relevant) typo